_posts/en/newsletters/2019-02-19-newsletter.md

```diff
@@ -0,0 +1,252 @@
+---
+title: 'Bitcoin Optech Newsletter #34'
+permalink: /en/newsletters/2019/02/19/
+name: 2019-02-19-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter summarizes a discussion about output tagging for
+BIP118 SIGHASH_NOINPUT_UNSAFE, announces merges that will make it
+possible to pair Bitcoin Core's built-in wallet in watching-only mode
+with a hardware wallet, and describes the completion of the feature
+freeze for the next Bitcoin Core release.  Also summarized are numerous
+code and documentation changes in popular Bitcoin infrastructure
+projects.
+
+## Action items
+
+None this week.
+
+## News
+
+- **Discussion about tagging outputs to enable restricted features on
+  spending:** The [BIP118][] SIGHASH_NOINPUT_UNSAFE (noinput) proposal
+  allows the person generating a signature that authorizes the spend of
+  one UTXO to optionally allow that signature to be reused ("replayed")
+  for spending other UTXOs sent to the same public key.  This enables
+  new features when used with something like payment channels that all
+  contain the same public keys (see the proposed [Eltoo][] layer for
+  LN), but it also makes possible *replay attacks* that can result in a
+  loss of money when users reuse addresses.  For example: Alice uses a
+  noinput signature when spending some money she previously received to
+  one of her addresses.  Later someone else pays that same address
+  intending to send Alice some more money.  Now anyone with a copy of
+  Alice's earlier signature from the block chain can send that money to
+  the same person Alice paid earlier by replaying Alice's earlier
+  signature.
+
+    One way to prevent such accidents is simply by appending "UNSAFE" to
+    the name of the feature, encouraging developers to learn about the
+    protocol's nuances before they implement it in their tools.
+    However, some developers have been looking for additional ways to
+    prevent problems.  In December, Johnson Lau [proposed][lau output
+    tagging] only allowing noinput to be used if the output being spent
+    had been specially tagged at its creation to allow the use of
+    noinput.  This would only allow the feature to be used when both the
+    spender and receiver agreed (as in the case of a payment channel),
+    preventing any miscommunications or misunderstandings from ending in
+    loss of funds.
+
+    Renewed discussion [last week][nick output tagging] and this week
+    saw analysis of the impact this would have on proposed layer two
+    protocols such as Eltoo and [Channel Factories][].  Although the
+    proposal obviously increases complexity, the general opinion seems
+    to be that it doesn't fundamentally increase the cost or reduce the
+    effectiveness of the described proposals, although it could make
+    them a bit less private (because you'd communicate that noinput was
```

Comment:
Suggestion: remove the parenthesized sentence. I think it's too much detail for this summary.

Reply:
Are we certain noinput is communicated as an option in the case it isn't used (cooperative closing of channel)? John asks for less detail, but I actually want more to understand why this is the case :) Perhaps you can link to where this is explained?

Reply:
For eltoo, it seems like you'd only use a noinput tagged output if you were doing a non-cooperative close, which will make it pretty clear you're using eltoo anyway. You're not necessarily *using* noinput in that case -- you only need to use a noinput signature if someone's actually playing an old state -- but the privacy is already gone because you are using clever eltoo scripting.

jnewbery's suggestion of just removing the aside seems reasonable to me fwiw

Reply:
Ok but I'm not seeing how it makes it less private, so unless I'm mistaken, I don't think we should state it makes "them a bit less private." In the cooperative close case, there is no privacy loss. In the non-cooperative close case, as you point out, privacy is already leaked because of the eltoo structure, so the addition of NOINPUT doesn't represent an incremental privacy leak.

Reply:
There might be non-eltoo protocols where you want to be a tx on-chain in the normal case, that also needs to be spendable by a noinput signature; noinput tagging would be less private in that case. We haven't thought up any protocols like that yet, but they might exist.